### PR TITLE
support PEP660 pip install --editable

### PR DESCRIPTION
**Pull request by Thirdegree**
implements #14 

~~Obviously WIP, as the https://github.com/pypa/pip/pull/8212 is still WIP~~ pip 21.3 is released, with this feature.

I'm pretty happy with this, it's for sure much cleaner than the whole "dump json to tmp file, write setup.py toload that json again, pip install -e"

I did test with the lnked project and it worked exactly how I'd expect an editable install to work so that's good.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -102,7 +103,7 @@ def test_ep() -> None:
 [project]
 name = "testproject"
 description = "an example test project for testing vulcan builds, %"
-authors = [{name="Joel Christiansen", email="joelchristiansen@optiver.com"}]
+authors = [{{name="Joel Christiansen", email="joelchristiansen@optiver.com"}}]
```
Comment (wisepotato) [inline tests/conftest.py:106]: why?

Comment (Thirdegree) [reply tests/conftest.py:106]: This is now being `.format`ed, so the literal curly braces need to be escaped, which is done by doubling up

File: vulcan/build_backend.py [hunk 2]
```diff
@@ -121,58 +128,127 @@ def get_virtualenv_python() -> Path:
...
+    m = re.search(r'pip (\d+\.\d+(\.\d+)?)', out)
+    if not m:
+        return None
+    return tuple((int(n) for n in m.group(1).split('.')))
+
+
+@contextmanager
+def maybe_gen_setuppy(venv: Path, config: Vulcan) -> Generator[None, None, None]:
+    pip_version = get_pip_version(venv)
+    if pip_version is None or pip_version < (999, 999, 999):  # TODO: Correct version once released
```
[the remark this answers is not in the document]
Comment (Thirdegree) [reply vulcan/build_backend.py:151]: Will also need to specify the minimum pip version in the tox, and ideally figure out a way to also test the old version...

**Comment by Thirdegree** [comment]
Linked pip PR has been merged, so this can be updated. Milestone for PIP is 21.3.

**Pull request merged**